win32ss/user/user32/windows/mdi.c

```diff
@@ -2047,6 +2047,27 @@ GetCascadeChildProc(HWND hwnd, LPARAM lParam)
     return TRUE;
 }
 
+static void
+QuerySizeFix(HWND hwnd, LPINT pcx, LPINT pcy)
+{
+    MINMAXINFO mmi;
+    DWORD dwResult;
+
+    mmi.ptMinTrackSize.x = mmi.ptMinTrackSize.y = 0;
+    mmi.ptMaxTrackSize.x = mmi.ptMaxTrackSize.y = MAXLONG;
+    SendMessageTimeoutW(hwnd, WM_GETMINMAXINFO, 0, (LPARAM)&mmi,
+                        SMTO_ABORTIFHUNG, 80, &dwResult);
```

Comment:
How should we handle a window that is hanging?
On windows, you can test this by adding a `Sleep(40 * 1000)` in the `WM_GETMINMAXINFO` handler. (Or any other timeout)

Reply:
I tried with WM_COMMAND. If timeout has come, SendMessageTimeout returns zero.